test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -49,9 +50,12 @@ pick_a_version()
     #     version 1.8.17 (installed on the build hosts). WT-7437 was introduced since 4.4.7.
     mapfile -t versions < <( git tag | grep $branch | grep -Ev "rc|alpha|4.4.[0-6]$" )
 
-    # Randomly pick a version from the array of patch versions
-    pv=${versions[$RANDOM % ${#versions[@]} ]}
-    echo "$pv"
+    len_versions=${#versions[@]}
+    if [ $len_versions != 0 ]; then
```

Comment:
Is that behaviour expected? Don't we want to check each branch? If we are ok with not testing all of them, we can leave it as is or print a message. @lukech

Reply:
I think this section of changes is trying to address the case that currently `mongodb-6.0` branch only has RC (and alpha) but no GA versions tagged. This function is expected to randomly pick a GA version from the given branch. 

We should have an "else" statement (after line 58) to fall back on picking an RC version since no GA version is available. If neither GA nor RC version is available we probably should error out - i.e the branch should not be added into this compatibility test.

Reply:
I like the idea of trying harder to find a branch we can test. By removing the `rc` bit and what's not relevant to newer branches, we could end up with the following to find a RC branch:

```
git tag | grep $branch | grep -Ev "alpha"
```

We could also try harder and test alpha branches but I am not sure if we want this.

Reply:
Actually, I had another thought for the use case - this `pick_a_version()` function is only called by the patch release compatibility test, which is not critical for the current stage of the `mongodb-6.0` release branch (leading up to the 1st GA version). As long as other types of compatibility tests ("-n" and "-u") cover the `mongodb-6.0` branch (which I believe they have) it should be fine. See please ignore the suggestions in my previous comment.